### PR TITLE
[Supplier] feat: add `staking_fee` param to supplier module

### PR DESCRIPTION
**Pull request by bryanchriswhite**
## Summary

Add `staking_fee` param to the tokenomics module.

## Issue

- `TODO_BETA`

## Type of change

Select one or more from the following:

- [x] New feature, functionality or library
- [ ] Consensus breaking; add the `consensus-breaking` label if so. See #791 for details
- [ ] Bug fix
- [ ] Code health or cleanup
- [ ] Documentation
- [ ] Other (specify)

## Testing

- [ ] **Documentation**: `make docusaurus_start`; only needed if you make doc changes
- [x] **Unit Tests**: `make go_develop_and_test`
- [ ] **LocalNet E2E Tests**: `make test_e2e`
- [x] **DevNet E2E Tests**: Add the `devnet-test-e2e` label to the PR.

## Sanity Checklist

- [x] I have tested my changes using the available tooling
- [ ] I have commented my code
- [x] I have performed a self-review of my own code; both comments & source code
- [ ] I create and reference any new tickets, if applicable
- [ ] I have left TODOs throughout the codebase, if applicable

### CODE REVIEW

File: testutil/integration/suites/param_configs.go [hunk 1]
```diff
@@ -70,8 +68,8 @@ var (
 	ValidProofMissingPenaltyCoin       = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 500)
 	ValidProofSubmissionFeeCoin        = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 5000000)
 	ValidProofRequirementThresholdCoin = cosmostypes.NewInt64Coin(volatile.DenomuPOKT, 100)
-	ValidRelayDifficultyTargetHash, _  = hex.DecodeString("00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff")
```
Comment (bryanchriswhite) [inline testutil/integration/suites/param_configs.go:73]: @reviewer, this was unused.

File: tools/scripts/params/supplier_staking_fee.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/poktroll.supplier.MsgUpdateParam",
+        "authority": "pokt10d07y265gmmuvt4z0w9aw880jnsr700j8yv32t",
+        "name": "staking_fee",
+        "as_coin": {
+          "amount": "1000000",
```
Comment (bryanchriswhite) [inline tools/scripts/params/supplier_staking_fee.json:9]:
```suggestion
          "amount": "1",
```

File: x/supplier/types/params.go [hunk 3]
```diff
@@ -14,6 +14,10 @@ var (
 	ParamMinStake = "min_stake"
 	// TODO_MAINNET: Determine the default value.
 	DefaultMinStake = cosmostypes.NewInt64Coin("upokt", 1000000) // 1 POKT
+	KeyStakingFee   = []byte("StakingFee")
+	ParamStakingFee = "staking_fee"
+	// TODO_MAINNET: Determine the default value.
+	DefaultStakingFee = cosmostypes.NewInt64Coin("upokt", 1000000) // 1 POKT
```
Comment (bryanchriswhite) [inline x/supplier/types/params.go:20]:
```suggestion
	DefaultStakingFee = cosmostypes.NewInt64Coin("upokt", 1) // 1 uPOKT
```

File: x/supplier/types/message_update_param.go [hunk 4]
```diff
@@ -45,23 +45,27 @@ func (msg *MsgUpdateParam) ValidateBasic() error {
...
+		return ValidateStakingFee(msg.GetAsCoin())
 	default:
 		return ErrSupplierParamInvalid.Wrapf("unsupported param %q", msg.Name)
 	}
 }
 
-// paramTypeIsCoin checks if the parameter type is a Coin, returning an error if not.
-func (msg *MsgUpdateParam) paramTypeIsCoin() error {
-	if _, ok := msg.AsType.(*MsgUpdateParam_AsCoin); !ok {
+func genericParamTypeIs[T any](msg *MsgUpdateParam) error {
```
Comment (red-0ne) [inline x/supplier/types/message_update_param.go:62]: Could you please add a godoc here?

File: x/supplier/types/params.go [hunk 5]
```diff
@@ -50,30 +66,57 @@ func (p Params) Validate() error {
...
+// ValidateStakingFee validates the StakingFee param.
+func ValidateStakingFee(stakingFeeAny any) error {
+	if _, err := paramAsPositiveuPOKT(stakingFeeAny, "staking_fee"); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO_IN_THIS_COMMOT: godoc...
```
Comment (red-0ne) [inline x/supplier/types/params.go:99]: Please add godoc before merging.

File: x/supplier/types/params.go [hunk 6]
```diff
@@ -50,30 +66,57 @@ func (p Params) Validate() error {
...
+func paramAsPositiveuPOKT(paramAny any, paramName string) (*cosmostypes.Coin, error) {
```
Comment (red-0ne) [inline x/supplier/types/params.go:100]:
```suggestion
func paramAsStrictlyPositiveuPOKT(paramAny any, paramName string) (*cosmostypes.Coin, error) {
```

File: x/supplier/keeper/params_test.go [hunk 7]
```diff
@@ -60,3 +71,44 @@ func TestParams_ValidateMinStake(t *testing.T) {
...
+			},
+			expectedErr: suppliertypes.ErrSupplierParamInvalid.Wrapf(
+				"staking_fee amount must be positive: got -1%s",
+				volatile.DenomuPOKT,
+			),
+		},
+		{
+			desc:     "valid SupplierStakingFee",
+			minStake: &suppliertypes.DefaultStakingFee,
+		},
```
Comment (red-0ne) [inline x/supplier/keeper/params_test.go:100]: If we're intentionally allowing a `0` staking fee, we should have a test for it.

**Review by red-0ne — COMMENTED** [review]
Left a few comments but nothing major.

File: x/supplier/types/params.go (hunk 6 above)
Comment (bryanchriswhite) [reply x/supplier/types/params.go:100]: The distinction is not clear to me. Can you elaborate and/or share your rationale?

Comment (red-0ne) [reply x/supplier/types/params.go:101]: My bad, I see it allows zero values.

**Review by red-0ne — APPROVED** [review]
LGTM

**Pull request merged**